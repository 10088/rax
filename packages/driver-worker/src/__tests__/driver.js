import Rax from 'rax';
import createDriver from '../index';

describe('driver-worker', () => {
  it('should postMessage by driver', (done) => {
    const addEventListener = () => {};
    const postMessage = (message) => {
      expect(message).toMatchSnapshot();
      done();
    };
    const driver = createDriver({ addEventListener, postMessage });
    Rax.render(Rax.createElement('view'), null, { driver });
  });
});
