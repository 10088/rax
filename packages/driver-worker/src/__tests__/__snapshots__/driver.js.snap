// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`driver-worker should postMessage by driver 1`] = `
Object {
  "mutations": Array [
    Object {
      "addedNodes": Array [
        Object {
          "$$id": "1",
          "attributes": Array [],
          "nodeName": "VIEW",
          "nodeType": 1,
        },
      ],
      "target": Object {
        "$$id": "2",
        "nodeName": "BODY",
      },
      "type": "childList",
    },
  ],
  "type": "MutationRecord",
}
`;
